lib/Controller/WizardController.php

```diff
@@ -0,0 +1,185 @@
+<?php
+/**
+ * @author Jörn Friedrich Dreyer <jfd@butonic.de>
+ *
+ * @copyright Copyright (c) 2018, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\User_LDAP\Controller;
+
+use OCA\User_LDAP\Access;
+use OCA\User_LDAP\Configuration;
+use OCA\User_LDAP\Connection;
+use OCA\User_LDAP\LDAP;
+use OCA\User_LDAP\User\Manager;
+use OCA\User_LDAP\Wizard;
+use OCA\User_LDAP\WizardResult;
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\IL10N;
+use OCP\IRequest;
+
+/**
+ * Class WizardController
+ *
+ * @package OCA\User_LDAP\Controller
+ */
+class WizardController extends Controller {
+
+	/** @var Manager */
+	protected $manager;
+	/** @var IL10N */
+	protected $l10n;
+	/** @var LDAP */
+	protected $ldapWrapper;
+
+	/**
+	 * @param string $appName
+	 * @param IRequest $request
+	 * @param IL10N $l10n
+	 * @param LDAP $ldapWrapper
+	 */
+	public function __construct($appName,
+								IRequest $request,
+								Manager $manager,
+								IL10N $l10n,
+								LDAP $ldapWrapper
+	) {
+		parent::__construct($appName, $request);
+		$this->manager = $manager;
+		$this->l10n = $l10n;
+		$this->ldapWrapper = $ldapWrapper;
+	}
+
+	/**
+	 * @param string $prefix config id
+	 * @param Configuration $configuration
+	 * @return Wizard
+	 */
+	private function getWizard($prefix, Configuration $configuration) {
+		$con = new Connection($this->ldapWrapper, $prefix, null);
+		$con->setConfiguration($configuration->getConfiguration());
+		$con->ldapConfigurationActive = true;
+		$con->setIgnoreValidation(true);
+
+		$access = new Access($con, $this->manager);
+
+		return new Wizard($configuration, $this->ldapWrapper, $access);
+	}
+
+	/**
+	 * do your magic
+	 *
+	 * @param string $ldap_serverconfig_chooser config id
+	 * @param string $action the spell to cast
+	 * @param string $cfgkey optional
+	 * @param string $cfgval optional
+	 * @param string $ldap_test_loginname optional
+	 * @return DataResponse
+	 */
+	public function cast($ldap_serverconfig_chooser, $action, $cfgkey = null, $cfgval = null, $ldap_test_loginname = null) {
+		$prefix = $ldap_serverconfig_chooser;
```

Comment:
needed?

Reply:
for now. I don't want to touch js with this PR. will remove in a subsequent pr. hopefully ... depends on how it affects js